### PR TITLE
avc/hevc: add max min frame qp test for cbr/vbr encode

### PR DESCRIPTION
**Pull request by zhuqingliang**
Signed-off-by: Zhu Qingliang <qingliangx.zhu@intel.com>

### CODE REVIEW

**Comment by zhoujd** [comment]
It looks good to me.

File: test/ffmpeg-qsv/encode/encoder.py [hunk 1]
```diff
@@ -55,6 +55,10 @@ def gen_output_opts(self):
     if vars(self).get("ladepth", None) is not None:
       opts += " -look_ahead 1"
       opts += " -look_ahead_depth {ladepth}"
+    if vars(self).get("qmax", None) is not None:
+      opts += " -qmax {qmax}"
+    if vars(self).get("qmin", None) is not None:
+      opts += " -qmin {qmin}"
```
Comment (uartie) [inline test/ffmpeg-qsv/encode/encoder.py]: qmin and qmax should be handled in gen_name, too.

File: test/ffmpeg-vaapi/encode/encoder.py [hunk 2]
```diff
@@ -65,6 +65,10 @@ def gen_output_opts(self):
     if vars(self).get("level", None) is not None:
       self.level /= 10.0
       opts += " -level {level}"
+    if vars(self).get("qmax", None) is not None:
+      opts += " -qmax {qmax}"
+    if vars(self).get("qmin", None) is not None:
+      opts += " -qmin {qmin}"
```
Comment (uartie) [inline test/ffmpeg-vaapi/encode/encoder.py]: qmax and qmin should be handled in gen_name, too.

File: test/gst-vaapi/encode/encoder.py [hunk 3]
```diff
@@ -63,6 +63,10 @@ def gen_output_opts(self):
       opts += " sharpness-level={loopshp}"
     if vars(self).get("looplvl", None) is not None:
       opts += " loop-filter-level={looplvl}"
+    if vars(self).get("qmax", None) is not None:
+      opts += " max-qp={qmax}"
+    if vars(self).get("qmin", None) is not None:
+      opts += " min-qp={qmin}"
```
Comment (uartie) [inline test/gst-vaapi/encode/encoder.py]: same as previous comments.

File: test/ffmpeg-qsv/encode/hevc.py [hunk 4]
```diff
@@ -89,7 +89,8 @@ def test_r2r(self, case, gop, slices, qp, quality, profile):
     self.encode()
 
 class cbr(HEVC8EncoderTest):
-  def init(self, tspec, case, gop, slices, bframes, bitrate, fps, profile):
+  def init(self, tspec, case, gop, slices, bframes, bitrate, fps, profile,
+          qmax = None, qmin = None):
```
Comment (uartie) [inline test/ffmpeg-qsv/encode/hevc.py]: AFAICT, the hevc encoder in ffmpeg-qsv does not consider/handle qmin or qmax.  Please correct me if I'm wrong.

Comment (zhuqingliang) [reply test/ffmpeg-qsv/encode/hevc.py]: Can we add  test feature in vaapi-fits , then to file a bug to developer ? I won't add reference for all test case.

Comment (uartie) [reply test/ffmpeg-qsv/encode/hevc.py]: No.  If middleware does not support the feature, then a test for the feature should not be added.  Please consider the potential that there may be other users of this test framework.  And if we add the test, then those users could generate their own cases and baselines and erroneously assume they are correct.

If you believe the middleware should support this feature, please file a bug in the middleware.

**Pull request closed without merging**